pydantic/types.py

```diff
@@ -516,3 +528,41 @@ def validate(cls, value: Union[str, bytes, int]) -> Union[IPv4Address, IPv6Addre
 
         with change_exception(errors.IPvAnyAddressError, ValueError):
             return IPv6Address(value)
+
+
+class IPvAnyInterface(_BaseAddress):
+    @classmethod
+    def __get_validators__(cls) -> 'CallableGenerator':
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, value: Union[str, bytes, int, NetworkType]) -> Union[IPv4Interface, IPv6Interface]:
+        try:
+            return IPv4Interface(value)
+        except ValueError:
+            pass
+
+        with change_exception(errors.IPvAnyInterfaceError, ValueError):
+            return IPv6Interface(value)
+
+
+class IPvAnyNetwork(_BaseNetwork):  # type: ignore
+    field: str = 'ip_vany_network_strict'
```

Comment:
I don't think we can implement `strict` like this, it's unlike anything else we do and will cause confusion.

I think that the default (`strict=True`) is fine. If people really want `strict=False` they can implement `IPvAnyNetworkLax` themselves.

Reply:
I've just fixed the ``strict`` argument